### PR TITLE
add more check for verifyDaemonSettings

### PR DESCRIPTION
**Pull request by hqhq**
Signed-off-by: Qiang Huang h.huangqiang@huawei.com

### CODE REVIEW

**Comment by hqhq** [comment]
I was thinking about combine this with `daemon.verifyHostConfig()`, seems `daemon.verifyHostConfig()` does a lot of thing which `verifyDaemonSettings` doesn't need to do. Let's keep them separate for now.

**Comment by jfrazelle** [comment]
seems sane to warn users again what they dont support, idk if its overkill, but i dont really see any harm LGTM

File: daemon/container.go [hunk 1]
```diff
@@ -1275,19 +1277,31 @@ func (container *Container) initializeNetworking() error {
...
-func (container *Container) verifyDaemonSettings() {
-	if container.hostConfig.Memory > 0 && !container.daemon.sysInfo.MemoryLimit {
-		logrus.Warnf("Your kernel does not support memory limit capabilities. Limitation discarded.")
-		container.hostConfig.Memory = 0
+// Make sure the config is compatible with the current kernel and daemon
+func (container *Container) verifyDaemonSettings() error {
+	hostConfig := container.hostConfig
+	daemon := container.daemon
+
+	if hostConfig.LxcConf.Len() > 0 && !strings.Contains(daemon.ExecutionDriver().Name(), "lxc") {
```
Comment (LK4D4) [inline daemon/container.go]: This checked in `verifyHostConfig`.

Comment (hqhq) [inline daemon/container.go]: @LK4D4 Yes, but it's related to daemon, if we created a container and passed this check, then we restart daemon with different exec driver, there would be a problem.
@jfrazelle Actually it's necessary, we could create a container, then change kernel and daemon setting and try to run this container. So daemon and kernel setting should always be checked when start containers.

Comment (Mashimiao) [inline daemon/container.go]: @hqhq, when starting containers, it really need check settings. But ContainerStart() already calls verifyHostConfig do it before calling container.Start().
So, there is not necessary to do double check

Comment (hqhq) [inline daemon/container.go]: Seems true, hostConfig checking used to only be done when create, not start, codes update so fast these days.

**Comment by hqhq** [comment]
verifyDaemonSettings seems not necessary for now. close it.

**Pull request closed without merging**